internal/table/chain.go

```diff
@@ -78,20 +78,49 @@ func (s *Chain) InstanceName() string {
 }
 
 func (s *Chain) Lookup(ctx context.Context, key string) (string, bool, error) {
+	newVal, err := s.LookupMulti(ctx, key)
+	if err != nil {
+		return "", false, err
+	}
+	if len(newVal) == 0 {
+		return "", false, nil
+	}
+
+	return newVal[0], true, nil
+}
+
+func (s *Chain) LookupMulti(ctx context.Context, key string) ([]string, error) {
+	result := []string{key}
+	STEP:
 	for i, step := range s.chain {
-		val, ok, err := step.Lookup(ctx, key)
-		if err != nil {
-			return "", false, err
-		}
-		if !ok {
-			if s.optional[i] {
-				continue
+		new_result := []string{}
+		for _, key = range result {
+			if step_multi, ok := step.(module.MultiTable); ok {
+				val, err := step_multi.LookupMulti(ctx, key)
+				if err != nil {
+					return []string{}, err
+				}
+				if len(val) == 0 {
```

Comment:
That probably should check s.optional in a way similar to step.Lookup branch below.

Reply:
good catch - done